tests/integration/pilot/workloadentry_test.go

```diff
@@ -0,0 +1,157 @@
+			ist, err := istio.Get(t)
+			if err != nil {
+				t.Fatal(err)
+			}
+			clusterCfg := t.Clusters().Default()
+			namespaceName := apps.Namespace.Name()
+
+			// Create a echo deployment "z" in the echos namespace.
+			t.Logf("Deploy an echo instance in namespace %s on cluster %s", namespaceName, clusterCfg.Name())
+			deployment.New(t, clusterCfg).
+				WithConfig(echo.Config{
+					Service:        "z",
+					ServiceAccount: true,
+					Namespace:      apps.Namespace,
+					Ports:          ports.All(),
+					Subsets:        []echo.SubsetConfig{{}},
+				}).BuildOrFail(t)
+
+			// Define an AUTO_PASSTHROUGH EW gateway
+			gatewayCfg := `apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: ingress-ew
+  namespace: istio-system
+spec:
+  selector:
+    istio: eastwestgateway
+  servers:
+  - port:
+      number: 15443
+      name: https
+      protocol: TLS
+    hosts:
+    - serviceentry.mesh.global
+    tls:
+      mode: AUTO_PASSTHROUGH
+`
+			// Configure an AUTO_PASSTHROUGH EW gateway
+			t.ConfigIstio().YAML("istio-system", gatewayCfg).ApplyOrFail(t, apply.CleanupConditionally)
+
+			ewGatewayIP, ewGatewayPort := ist.EastWestGatewayFor(clusterCfg).AddressForPort(15443)
+			if ewGatewayIP == "" || ewGatewayPort == 0 { // most likely EW gateway is not deployed, skip testing
+				t.Skipf("Skipping test, eastwest gateway is probably not deployed for cluster %s", clusterCfg.Name())
+			}
+
+			// Define ServiceEntry & WorkloadEntry for test
+			workloadEntryYaml := fmt.Sprintf(`apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+  name: z-se
+spec:
+  addresses:
+  - 240.240.34.56
+  hosts:
+  - serviceentry.mesh.global
+  ports:
+  - name: http
+    number: 80
+    protocol: HTTP
+    targetPort: 8080
+  location: MESH_INTERNAL
+  resolution: STATIC
+  workloadSelector:
+    labels:
+      app: z
+---
+apiVersion: networking.istio.io/v1beta1
+kind: WorkloadEntry
+metadata:
+  name: z-we
+  labels:
+    security.istio.io/tlsMode: istio
+    app: z
+spec:
+  ports:
+    http: %v
+  address: %s`, ewGatewayPort, ewGatewayIP)
+
+			// Configure ServiceEntry & WorkloadEntry
+			t.ConfigIstio().YAML(namespaceName, workloadEntryYaml).ApplyOrFail(t, apply.CleanupConditionally)
+
+			srcs := apps.All.Instances()
+			for _, src := range srcs {
+				srcName := src.Config().NamespacedName().Name
+				// Skipping tests for these workloads:
+				//      external
+				//      naked
+				//      proxyless-grpc
+				//      vm
+				if srcName == commonDeployment.ProxylessGRPCSvc ||
+					srcName == commonDeployment.NakedSvc ||
+					srcName == commonDeployment.ExternalSvc ||
+					srcName == commonDeployment.VMSvc {
+					continue
+				}
+				srcCluster := src.Config().Cluster.Name()
+				// Assert that non-skipped workloads can reach the service which includes our workload entry
+				t.NewSubTestf("%s in %s to ServiceEntry+WorkloadEntry Responds with 200", srcName, srcCluster).Run(func(t framework.TestContext) {
```

Comment:
I don't think this asserts we actually use the WE at all. maybe need check.ReachedTargetClusters?

Reply:
Part of the reason I settled on using the new deployment `z` is to ensure the destination I want to hit is only available in a single cluster. This (partially) prevents a scenario where `a` was reached but the `a` we got to was actually local and thus WE was not really tested. It's still possible that some tests reach the primary cluster from the remote without using the EW gateway during flat network multi-cluster scenarios so the test topologies that include a cross-network-primary cluster provide the strongest assertion that WE is working since the serviceentry has nothing other than the workload entry to select in this topology.

Reply:
Trying to grok this...


Ok so we deploy Z in "default cluster"
We get EW GW of "default cluster"
We set WE to point to EW GW of "default cluster"
We call the service from all clusters

Path expected is:
* Same cluster: ???
* Diff cluster, same network: ???
* Diff network: ???


I think we should actually assert the path, not just that it gets 200s. I don't actually understand how this works -- wouldn't we just be round-robining between the WE and pod?

Reply:
>Trying to grok this...

No worries at all. These are good questions and I appreciate you doublechecking to ensure the test is meaningful.

>Path expected is:

Same cluster: WE and pod, like you noted.
Diff cluster, same network: WE and pod.
Diff network: WE only (direct to pod isn't available for these src workloads, explained in more detail below).

>I think we should actually assert the path.

On the cross-network-primary the envoy cluster for this SE from a sidecar in `deploy/a` looks like this where `172.18.7.111` is the external IP of the EW gateway:

```
outbound|80||serviceentry.mesh.global::observability_name::outbound|80||serviceentry.mesh.global
outbound|80||serviceentry.mesh.global::default_priority::max_connections::4294967295
outbound|80||serviceentry.mesh.global::default_priority::max_pending_requests::4294967295
outbound|80||serviceentry.mesh.global::default_priority::max_requests::4294967295
outbound|80||serviceentry.mesh.global::default_priority::max_retries::4294967295
outbound|80||serviceentry.mesh.global::high_priority::max_connections::1024
outbound|80||serviceentry.mesh.global::high_priority::max_pending_requests::1024
outbound|80||serviceentry.mesh.global::high_priority::max_requests::1024
outbound|80||serviceentry.mesh.global::high_priority::max_retries::3
outbound|80||serviceentry.mesh.global::added_via_api::true
outbound|80||serviceentry.mesh.global::eds_service_name::outbound|80||serviceentry.mesh.global
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::cx_active::2
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::cx_connect_fail::0
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::cx_total::2
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::rq_active::0
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::rq_error::0
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::rq_success::3
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::rq_timeout::0
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::rq_total::3
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::hostname::
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::health_flags::healthy
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::weight::2
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::region::
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::zone::
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::sub_zone::
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::canary::false
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::priority::0
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::success_rate::-1
outbound|80||serviceentry.mesh.global::172.18.7.111:15443::local_origin_success_rate::-1
```

The traffic has no choice but to traverse the EW gateway.

Reply:
Additionally, this is only the case if the dst workload doesn't exist in the local network. If the dst workload is available in this network it will be discovered and configured as directly reachable. This circles back around to why I ended up deploying `z` rather than using an existing workload.